### ISSUE TITLE
SteamDeck: X-Morph (gog windows) does not start when called from steam

### ISSUE THREAD

**Issue by sluedecke** [Heroic-Games-Launcher/HeroicGamesLauncher#2136]
### Describe the bug

I've installed some GOG Windows games using 2.4.3 to my deck and added them to steam.  They worked fine when being called from steam in (big picture? non-desktop?) normal mode.

I revisited them recently and at least X-Morph:Defense Complete Edition appears to be broken.  It starts fine from desktop mode, but does not start from normal mode.  Abort game does has no effect, steam button -> exit game has no effect not work. The logs state that some dlls could not be imported.

### Add logs

```text
System Info:
Heroic Version: 2.5.0 Yamato
Legendary Version:  0.20.31 Dark Energy (hotfix #5)
GOGdl Version: 0.5
OS: Freedesktop SDK KERNEL: 5.13.0-valve21.3-1-neptune ARCH: x64
CPU: AMD Custom APU 0405 @2.24 GOVERNOR: schedutil
RAM: Total: 14.49 GiB Available: 10.97 GiB
GRAPHICS: GPU0: VanGogh [AMD Custom GPU 0405] VRAM: 512MB  
PROTOCOL: wayland

Game Settings: {
	"autoInstallDxvk": false,
	"autoInstallVkd3d": false,
	"preferSystemLibs": false,
	"maxSharpness": 2,
	"nvidiaPrime": false,
	"enviromentOptions": [],
	"wrapperOptions": [],
	"savesPath": "",
	"showFps": false,
	"targetExe": "",
	"useGameMode": false,
	"language": "",
	"wineVersion": {
		"bin": "/home/deck/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-Proton7-31/bin/wine",
		"name": "Wine - Wine-GE-Proton7-31",
		"type": "wine",
		"lib": "/home/deck/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-Proton7-31/lib64",
		"lib32": "/home/deck/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-Proton7-31/lib",
		"wineserver": "/home/deck/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-Proton7-31/bin/wineserver",
		"wineboot": "/home/deck/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-Proton7-31/bin/wineboot"
	},
	"winePrefix": "/home/deck/Games/Heroic/Prefixes/X-Morph Defense Complete Edition",
	"eacRuntime": false,
	"battlEyeRuntime": false,
	"gogSaves": []
}

Game launched at: Sun Dec 04 2022 11:14:56 GMT+0100 (Central European Standard Time)

Launch Command: LD_LIBRARY_PATH=/home/deck/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-Proton7-31/lib64:/home/deck/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-Proton7-31/lib WINEPREFIX="/home/deck/Games/Heroic/Prefixes/X-Morph Defense Complete Edition" ORIG_LD_LIBRARY_PATH= GST_PLUGIN_SYSTEM_PATH_1_0=/home/deck/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-Proton7-31/lib64/gstreamer-1.0:/home/deck/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-Proton7-31/lib/gstreamer-1.0 WINEDLLPATH=/home/deck/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-Proton7-31/lib64/wine:/home/deck/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-Proton7-31/lib/wine /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/gogdl launch "/run/media/mmcblk0p1/gog/X-Morph Defense Complete Edition" 1135975449 --wine /home/deck/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-Proton7-31/bin/wine --platform windows --skip-version-check

Game Log:
wine: using kernel write watches (experimental).
wineserver: using server-side synchronization.
wine: RLIMIT_NICE is <= 20, unable to use setpriority safely
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
002c:fixme:winediag:LdrInitializeThunk wine-staging 7.0 is a testing version containing experimental patches.
002c:fixme:winediag:LdrInitializeThunk Please mention your exact version when filing bug reports on winehq.org.
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
ATTENTION: default value of option vk_xwayland_wait_ready overridden by environment.
ATTENTION: default value of option vk_xwayland_wait_ready overridden by environment.
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
0118:err:module:import_dll Library dxgi.dll (which is needed by L"C:\\windows\\system32\\d3d11.dll") not found
0118:err:module:import_dll Library d3d11.dll (which is needed by L"E:\\gog\\X-Morph Defense Complete Edition\\bin\\shooter_win64_release.exe") not found
0118:err:module:import_dll Library dxgi.dll (which is needed by L"E:\\gog\\X-Morph Defense Complete Edition\\bin\\shooter_win64_release.exe") not found
0118:err:module:LdrInitializeThunk Importing dlls for L"E:\\gog\\X-Morph Defense Complete Edition\\bin\\shooter_win64_release.exe" failed, status c0000135
Launch command: ['/home/deck/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-Proton7-31/bin/wine', '/run/media/mmcblk0p1/gog/X-Morph Defense Complete Edition/bin/shooter_win64_release.exe', '--skip-version-check']
```

### Steps to reproduce

1. Install game in desktop mode
2. add game to steam
3. start game in desktop mode
4. switch to normal mode
5. try to start game

### Expected behavior

Game launches normally

### System Information

- OS: steam deck latest
- Heroic Version: 2.5.0

**Comment by sluedecke** [comment]
What I tried unsuccessfully:

* downgrading heroic to 2.4.3 

What works:

* run from heroic in desktop mode
* run from steam in desktop mode

**Comment by flavioislima** [comment]
this games has Cloud saves? if yes, even if auto sync is disabled, you need to setup a path for it. 
I found this bug on my tests and might be the reason it is not working for you.

**Comment by sluedecke** [comment]
Indeed it does.  I had to enable it in the game settings and I had to set the save folder manually.  It appears to sync saves now, but it still starts in desktop mode only.

In 'non-desktop mode' it does not start and the log is `/home/deck/.var/app/com.heroicgameslauncher.hgl/config/heroic/GamesConfig/1135975449-lastPlay.log`:

```
System Info:
Heroic Version: 2.5.1 Yamato
Legendary Version:  0.20.31 Dark Energy (hotfix #5)
GOGdl Version: 0.5
OS: Freedesktop SDK KERNEL: 5.13.0-valve21.3-1-neptune ARCH: x64
CPU: AMD Custom APU 0405 @2.5 GOVERNOR: schedutil
RAM: Total: 14.49 GiB Available: 11.74 GiB
GRAPHICS: GPU0: VanGogh [AMD Custom GPU 0405] VRAM: 512MB  
PROTOCOL: wayland

Game Settings: {
	"autoInstallDxvk": false,
	"autoInstallVkd3d": false,
	"preferSystemLibs": false,
	"autoSyncSaves": true,
	"maxSharpness": 2,
	"nvidiaPrime": false,
	"enviromentOptions": [],
	"wrapperOptions": [],
	"savesPath": "",
	"showFps": false,
	"targetExe": "",
	"useGameMode": false,
	"language": "",
	"wineVersion": {
		"bin": "/home/deck/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-Proton7-31/bin/wine",
		"name": "Wine - Wine-GE-Proton7-31",
		"type": "wine",
		"lib": "/home/deck/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-Proton7-31/lib64",
		"lib32": "/home/deck/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-Proton7-31/lib",
		"wineserver": "/home/deck/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-Proton7-31/bin/wineserver",
		"wineboot": "/home/deck/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-Proton7-31/bin/wineboot"
	},
	"winePrefix": "/home/deck/Games/Heroic/Prefixes/X-Morph Defense Complete Edition",
	"eacRuntime": false,
	"battlEyeRuntime": false,
	"gogSaves": [
		{
			"name": "saves",
			"location": "/home/deck/Games/Heroic/Prefixes/X-Morph Defense Complete Edition/drive_c/users/deck/Documents/X-Morph Defense"
		}
	]
}

Game launched at: Mon Dec 12 2022 10:21:17 GMT+0100 (Central European Standard Time)

Launch Command: LD_LIBRARY_PATH=/home/deck/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-Proton7-31/lib64:/home/deck/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-Proton7-31/lib WINEPREFIX="/home/deck/Games/Heroic/Prefixes/X-Morph Defense Complete Edition" ORIG_LD_LIBRARY_PATH= GST_PLUGIN_SYSTEM_PATH_1_0=/home/deck/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-Proton7-31/lib64/gstreamer-1.0:/home/deck/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-Proton7-31/lib/gstreamer-1.0 WINEDLLPATH=/home/deck/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-Proton7-31/lib64/wine:/home/deck/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-Proton7-31/lib/wine /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/gogdl launch "/run/media/mmcblk0p1/gog/X-Morph Defense Complete Edition" 1135975449 --wine /home/deck/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-Proton7-31/bin/wine --platform windows --skip-version-check

Game Log:
wine: using kernel write watches (experimental).
wineserver: using server-side synchronization.
wine: RLIMIT_NICE is <= 20, unable to use setpriority safely
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
002c:fixme:winediag:LdrInitializeThunk wine-staging 7.0 is a testing version containing experimental patches.
002c:fixme:winediag:LdrInitializeThunk Please mention your exact version when filing bug reports on winehq.org.
002c:fixme:wineboot:read_tsc_frequency TSC frequency calibration failed, unstable TSC?time0 13315310479075223800 ns, time1 13315310479076322800 ns
002c:fixme:wineboot:read_tsc_frequency tsc2 - tsc0 3067596, tsc3 - tsc1 3077508
002c:fixme:wineboot:read_tsc_frequency freq0 2791261146 Hz, freq2 2800280254 Hz, error 3231 ppm
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
ATTENTION: default value of option vk_xwayland_wait_ready overridden by environment.
ATTENTION: default value of option vk_xwayland_wait_ready overridden by environment.
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
wine: using kernel write watches (experimental).
wine: overriding CPU configuration, 8 logical CPUs, host CPUs 0,1,2,3,4,5,6,7.
011c:err:module:import_dll Library dxgi.dll (which is needed by L"C:\\windows\\system32\\d3d11.dll") not found
011c:err:module:import_dll Library d3d11.dll (which is needed by L"E:\\gog\\X-Morph Defense Complete Edition\\bin\\shooter_win64_release.exe") not found
011c:err:module:import_dll Library dxgi.dll (which is needed by L"E:\\gog\\X-Morph Defense Complete Edition\\bin\\shooter_win64_release.exe") not found
011c:err:module:LdrInitializeThunk Importing dlls for L"E:\\gog\\X-Morph Defense Complete Edition\\bin\\shooter_win64_release.exe" failed, status c0000135
Launch command: ['/home/deck/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-Proton7-31/bin/wine', '/run/media/mmcblk0p1/gog/X-Morph Defense Complete Edition/bin/shooter_win64_release.exe', '--skip-version-check']
All processes exited
```

**Comment by sluedecke** [comment]
I found the solution: setting wine to proton is the key.  Now it starts in non-desktop mode and desktop mode!

Case closed :)
